BWAPI4J/src/main/java/org/openbw/bwapi4j/unit/ZergResourceDepotImpl.java

```diff
@@ -0,0 +1,174 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+//    Copyright (C) 2017-2018 OpenBW Team
+//
+//    This file is part of BWAPI4J.
+//
+//    BWAPI4J is free software: you can redistribute it and/or modify
+//    it under the terms of the Lesser GNU General Public License as published 
+//    by the Free Software Foundation, version 3 only.
+//
+//    BWAPI4J is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU General Public License for more details.
+//
+//    You should have received a copy of the GNU General Public License
+//    along with BWAPI4J.  If not, see <http://www.gnu.org/licenses/>.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package org.openbw.bwapi4j.unit;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.openbw.bwapi4j.Position;
+import org.openbw.bwapi4j.type.TechType;
+import org.openbw.bwapi4j.type.UnitType;
+import org.openbw.bwapi4j.type.UpgradeType;
+import org.openbw.bwapi4j.unit.UnitImpl.TrainingSlot;
+
+public abstract class ZergResourceDepotImpl extends BuildingImpl implements Organic, ResourceDepot, ResearchingFacility, Morphable, TrainingFacility {

```

Comment:
Instead of implementing `ResourceDepot`, it should implement `ZergResourceDepot` where `ZergResourceDepot` is an interface that extends `ResourceDepot`. Then, users can write `unit instanceof ZergResourceDepot` instead of `instance of ZergResourceDepotImpl`.

Reply:
I see your point and can add an interface, but if one wanted to implement shared methods common to `Hatchery`, `Lair`, `Hive`, they'd have to write them separately (assuming you don't keep `Hive` extending a `Lair`, `Lair` extending a `Hatchery`) or have an abstract class to inherit from, which is what's currently proposed. I think the only functionality really unique to `ZergResourceDepot` would be `getLarva()` and `researchBurrowing()`.